example_client/grpc_stateful_client.py

```diff
@@ -32,154 +32,250 @@ def printDebug(msg):
-
-parser.add_argument('--utterances', required=False, default=10, help='How many utterances to process from ark file. default 10')
-parser.add_argument('--samples', required=False, default=500, help='How many samples to process from each utterance file. default 500')
-parser.add_argument('--debug', required=False, default=0, help='Enabling debug prints. default 0')
-
-args = vars(parser.parse_args())
-
-global debug_mode
-debug_mode = args.get('debug')
-channel = grpc.insecure_channel("{}:{}".format(args['grpc_address'],args['grpc_port']))
-stub = prediction_service_pb2_grpc.PredictionServiceStub(channel)
-
-processing_times = np.zeros((0),int)
-
-model_input_path = args.get('model_input_path')
-print('Reading ark file {}'.format(model_input_path))
-ark_reader = ArchiveReader(model_input_path)
-
-model_score_path = args.get('model_score_path')
-print('Reading scores ark file {}'.format(model_score_path))
-ark_score = ArchiveReader(model_score_path)
-numberOfKeys = 0
-
-for key, obj in ark_reader:
-    printDebug("Input ark file data range {0}: {1}".format(key, obj.shape))
-    numberOfKeys += 1
-
-for key, obj in ark_score:
-    printDebug("Scores ark file data range {0}: {1}".format(key, obj.shape))
-
-scoreObjects = { k:m for k,m in ark_score }
-
-print('Start processing:')
-print('\tModel name: {}'.format(args.get('model_name')))
-
-SEQUENCE_START = 1
-SEQUENCE_END = 2
-sequence_id = 1020
-utterances_limit = int(args.get('utterances'))
-utterance = 0
-
-meanErrGlobal = 0.0
-for key, obj in ark_reader:
-    utterance += 1
-    if utterance > utterances_limit:
-        break
-
-    batch_size = obj.shape[0]
-    print('\n\tInput name: {}'.format(key))
-    printDebug('\tInput in shape: {}'.format(obj.shape))
-    printDebug('\tInput batch size: {}'.format(batch_size))
-    printDebug('\tSequence id: {}'.format(sequence_id))
+    if 'sequence_id' not in result.outputs:
+        print("ERROR: Missing sequence_id in model output")
+        print("Available outputs:")
+        for Y in result.outputs:
+            print(Y)
+        exit(1)
 
-    meanErrSum = 0.0
+def main():
+    args = ParseArguments()
+    global debug_mode
+    debug_mode = int(args.get('debug'))
+
+    channel = grpc.insecure_channel("{}:{}".format(args['grpc_address'],args['grpc_port']))
+    stub = prediction_service_pb2_grpc.PredictionServiceStub(channel)
+    processing_times = np.zeros((0),int)
+    cw_l = int(args.get('cw_l'))
+    cw_r = int(args.get('cw_r'))
+    print('\tContext window left width cw_l: {}'.format(cw_l))
+    print('\tContext window right width cw_r: {}'.format(cw_r))
+    print('Start processing:')
+    print('\tModel name: {}'.format(args.get('model_name')))
+
+    initial_ark_reader, ark_scores, input_names, output_names, numberOfKeys, inputArrays, referenceArrays = PrepareProcessingData(args)
+
+    SEQUENCE_START = 1
+    SEQUENCE_END = 2
+    sequence_id = 2
+    meanErrGlobal = 0.0
+
+    for key, obj in initial_ark_reader:
```

Comment:
Can we extract sequence name and sequence size for every sequence since each input and expected output will have the same number of sequences with the same names and the same number of requests/samples in it?